--- D:\CRAFTSTONE\DiffMakerPrivate\source\orig\net\minecraft\command\CommandShowSeed.java	2014-11-28 15:10:38.000000000 +0100
+++ D:\CRAFTSTONE\DiffMakerPrivate\source\new\net\minecraft\command\CommandShowSeed.java	2014-12-02 18:25:49.466395600 +0100
@@ -17,10 +17,6 @@
       return "seed";
    }
 
-   public int getRequiredPermissionLevel() {
-      return 2;
-   }
-
    public String getCommandUsage(ICommandSender var1) {
       return "commands.seed.usage";
    }
