--- D:\CRAFTSTONE\DiffMakerPrivate\source\orig\net\minecraft\command\CommandToggleDownfall.java	2014-11-28 15:10:38.000000000 +0100
+++ D:\CRAFTSTONE\DiffMakerPrivate\source\new\net\minecraft\command\CommandToggleDownfall.java	2014-12-02 18:26:08.197461400 +0100
@@ -1,8 +1,7 @@
 package net.minecraft.command;
 
-import net.minecraft.command.CommandBase;
-import net.minecraft.command.ICommandSender;
 import net.minecraft.server.MinecraftServer;
+import net.minecraft.world.WorldServer;
 import net.minecraft.world.storage.WorldInfo;
 
 public class CommandToggleDownfall extends CommandBase {
@@ -11,10 +10,6 @@
       return "toggledownfall";
    }
 
-   public int getRequiredPermissionLevel() {
-      return 2;
-   }
-
    public String getCommandUsage(ICommandSender var1) {
       return "commands.downfall.usage";
    }
@@ -25,7 +20,9 @@
    }
 
    protected void toggleDownfall() {
-      WorldInfo var1 = MinecraftServer.getServer().worldServers[0].getWorldInfo();
-      var1.setRaining(!var1.isRaining());
+	  for (WorldServer world : MinecraftServer.getServer().worldServers) {
+	      WorldInfo var1 = world.getWorldInfo();
+	      var1.setRaining(!var1.isRaining());
+	  }
    }
 }
