--- D:\CRAFTSTONE\DiffMakerPrivate\source\orig\net\minecraft\command\CommandGive.java	2014-11-28 15:10:38.000000000 +0100
+++ D:\CRAFTSTONE\DiffMakerPrivate\source\new\net\minecraft\command\CommandGive.java	2014-12-02 18:25:16.690304200 +0100
@@ -20,16 +20,12 @@
    public String getCommandName() {
       return "give";
    }
-
-   public int getRequiredPermissionLevel() {
-      return 2;
-   }
-
+   
    public String getCommandUsage(ICommandSender var1) {
       return "commands.give.usage";
    }
 
-   public void processCommand(ICommandSender var1, String[] var2) {
+   public void processCommand(ICommandSender var1, String[] var2) throws CommandException {
       if(var2.length < 2) {
          throw new WrongUsageException("commands.give.usage", new Object[0]);
       } else {
