--- D:\CRAFTSTONE\DiffMakerPrivate\source\orig\net\minecraft\command\server\CommandListBans.java	2014-11-28 15:10:38.000000000 +0100
+++ D:\CRAFTSTONE\DiffMakerPrivate\source\new\net\minecraft\command\server\CommandListBans.java	2014-12-02 18:26:55.746622100 +0100
@@ -14,10 +14,6 @@
       return "banlist";
    }
 
-   public int getRequiredPermissionLevel() {
-      return 3;
-   }
-
    public boolean canCommandSenderUseCommand(ICommandSender var1) {
       return (MinecraftServer.getServer().getConfigurationManager().getBannedIPs().isLanServer() || MinecraftServer.getServer().getConfigurationManager().getBannedPlayers().isLanServer()) && super.canCommandSenderUseCommand(var1);
    }
