--- D:\CRAFTSTONE\DiffMakerPrivate\source\orig\net\minecraft\command\CommandClone.java	2014-11-28 15:10:38.000000000 +0100
+++ D:\CRAFTSTONE\DiffMakerPrivate\source\new\net\minecraft\command\CommandClone.java	2014-12-02 18:24:36.793186400 +0100
@@ -27,15 +27,11 @@
       return "clone";
    }
 
-   public int getRequiredPermissionLevel() {
-      return 2;
-   }
-
    public String getCommandUsage(ICommandSender var1) {
       return "commands.clone.usage";
    }
 
-   public void processCommand(ICommandSender var1, String[] var2) {
+   public void processCommand(ICommandSender var1, String[] var2) throws CommandException {
       if(var2.length < 9) {
          throw new WrongUsageException("commands.clone.usage", new Object[0]);
       } else {
