--- D:\CRAFTSTONE\DiffMakerPrivate\source\orig\net\minecraft\command\server\CommandWhitelist.java	2014-11-28 15:10:40.000000000 +0100
+++ D:\CRAFTSTONE\DiffMakerPrivate\source\new\net\minecraft\command\server\CommandWhitelist.java	2014-12-02 18:27:51.124807200 +0100
@@ -17,15 +17,11 @@
       return "whitelist";
    }
 
-   public int getRequiredPermissionLevel() {
-      return 3;
-   }
-
    public String getCommandUsage(ICommandSender var1) {
       return "commands.whitelist.usage";
    }
 
-   public void processCommand(ICommandSender var1, String[] var2) {
+   public void processCommand(ICommandSender var1, String[] var2) throws CommandException {
       if(var2.length < 1) {
          throw new WrongUsageException("commands.whitelist.usage", new Object[0]);
       } else {
