--- D:\CRAFTSTONE\DiffMakerPrivate\source\orig\net\minecraft\command\CommandFill.java	2014-11-28 15:10:38.000000000 +0100
+++ D:\CRAFTSTONE\DiffMakerPrivate\source\new\net\minecraft\command\CommandFill.java	2014-12-02 18:25:05.571274200 +0100
@@ -26,15 +26,11 @@
       return "fill";
    }
 
-   public int getRequiredPermissionLevel() {
-      return 2;
-   }
-
    public String getCommandUsage(ICommandSender var1) {
       return "commands.fill.usage";
    }
 
-   public void processCommand(ICommandSender var1, String[] var2) {
+   public void processCommand(ICommandSender var1, String[] var2) throws CommandException {
       if(var2.length < 7) {
          throw new WrongUsageException("commands.fill.usage", new Object[0]);
       } else {
