--- D:\CRAFTSTONE\DiffMakerPrivate\source\orig\net\minecraft\command\CommandTrigger.java	2014-11-28 15:10:38.000000000 +0100
+++ D:\CRAFTSTONE\DiffMakerPrivate\source\new\net\minecraft\command\CommandTrigger.java	2014-12-02 18:26:12.637468800 +0100
@@ -23,15 +23,11 @@
       return "trigger";
    }
 
-   public int getRequiredPermissionLevel() {
-      return 0;
-   }
-
    public String getCommandUsage(ICommandSender var1) {
       return "commands.trigger.usage";
    }
 
-   public void processCommand(ICommandSender var1, String[] var2) {
+   public void processCommand(ICommandSender var1, String[] var2) throws CommandException {
       if(var2.length < 3) {
          throw new WrongUsageException("commands.trigger.usage", new Object[0]);
       } else {
