--- D:\CRAFTSTONE\DiffMakerPrivate\source\orig\net\minecraft\command\CommandBlockData.java	2014-11-28 15:10:36.000000000 +0100
+++ D:\CRAFTSTONE\DiffMakerPrivate\source\new\net\minecraft\command\CommandBlockData.java	2014-12-02 18:24:30.418166100 +0100
@@ -19,15 +19,11 @@
       return "blockdata";
    }
 
-   public int getRequiredPermissionLevel() {
-      return 2;
-   }
-
    public String getCommandUsage(ICommandSender var1) {
       return "commands.blockdata.usage";
    }
 
-   public void processCommand(ICommandSender var1, String[] var2) {
+   public void processCommand(ICommandSender var1, String[] var2) throws NumberInvalidException, PlayerNotFoundException, CommandException {
       if(var2.length < 4) {
          throw new WrongUsageException("commands.blockdata.usage", new Object[0]);
       } else {
