--- D:\CRAFTSTONE\DiffMakerPrivate\source\orig\net\minecraft\command\CommandParticle.java	2014-11-28 15:10:38.000000000 +0100
+++ D:\CRAFTSTONE\DiffMakerPrivate\source\new\net\minecraft\command\CommandParticle.java	2014-12-02 18:25:31.350345900 +0100
@@ -17,15 +17,11 @@
       return "particle";
    }
 
-   public int getRequiredPermissionLevel() {
-      return 2;
-   }
-
    public String getCommandUsage(ICommandSender var1) {
       return "commands.particle.usage";
    }
 
-   public void processCommand(ICommandSender var1, String[] var2) {
+   public void processCommand(ICommandSender var1, String[] var2) throws CommandException {
       if(var2.length < 8) {
          throw new WrongUsageException("commands.particle.usage", new Object[0]);
       } else {
