--- D:\CRAFTSTONE\DiffMakerPrivate\source\orig\net\minecraft\command\server\CommandPardonPlayer.java	2014-11-28 15:10:38.000000000 +0100
+++ D:\CRAFTSTONE\DiffMakerPrivate\source\new\net\minecraft\command\server\CommandPardonPlayer.java	2014-12-02 18:27:25.400712900 +0100
@@ -15,10 +15,6 @@
       return "pardon";
    }
 
-   public int getRequiredPermissionLevel() {
-      return 3;
-   }
-
    public String getCommandUsage(ICommandSender var1) {
       return "commands.unban.usage";
    }
@@ -27,7 +23,7 @@
       return MinecraftServer.getServer().getConfigurationManager().getBannedPlayers().isLanServer() && super.canCommandSenderUseCommand(var1);
    }
 
-   public void processCommand(ICommandSender var1, String[] var2) {
+   public void processCommand(ICommandSender var1, String[] var2) throws CommandException {
       if(var2.length == 1 && var2[0].length() > 0) {
          MinecraftServer var3 = MinecraftServer.getServer();
          GameProfile var4 = var3.getConfigurationManager().getBannedPlayers().isUsernameBanned(var2[0]);
