--- D:\CRAFTSTONE\DiffMakerPrivate\source\orig\net\minecraft\command\CommandEffect.java	2014-11-28 15:10:38.000000000 +0100
+++ D:\CRAFTSTONE\DiffMakerPrivate\source\new\net\minecraft\command\CommandEffect.java	2014-12-02 18:24:52.908234500 +0100
@@ -19,15 +19,11 @@
       return "effect";
    }
 
-   public int getRequiredPermissionLevel() {
-      return 2;
-   }
-
    public String getCommandUsage(ICommandSender var1) {
       return "commands.effect.usage";
    }
 
-   public void processCommand(ICommandSender var1, String[] var2) {
+   public void processCommand(ICommandSender var1, String[] var2) throws CommandException {
       if(var2.length < 2) {
          throw new WrongUsageException("commands.effect.usage", new Object[0]);
       } else {
