--- D:\CRAFTSTONE\DiffMakerPrivate\source\orig\net\minecraft\command\CommandPlaySound.java	2014-11-28 15:10:38.000000000 +0100
+++ D:\CRAFTSTONE\DiffMakerPrivate\source\new\net\minecraft\command\CommandPlaySound.java	2014-12-02 18:25:33.763350000 +0100
@@ -17,15 +17,11 @@
       return "playsound";
    }
 
-   public int getRequiredPermissionLevel() {
-      return 2;
-   }
-
    public String getCommandUsage(ICommandSender var1) {
       return "commands.playsound.usage";
    }
 
-   public void processCommand(ICommandSender var1, String[] var2) {
+   public void processCommand(ICommandSender var1, String[] var2) throws CommandException {
       if(var2.length < 2) {
          throw new WrongUsageException(this.getCommandUsage(var1), new Object[0]);
       } else {
