--- D:\CRAFTSTONE\DiffMakerPrivate\source\orig\net\minecraft\entity\passive\EntityPig.java	2014-11-28 15:10:48.000000000 +0100
+++ D:\CRAFTSTONE\DiffMakerPrivate\source\new\net\minecraft\entity\passive\EntityPig.java	2014-11-28 14:55:04.785311000 +0100
@@ -162,9 +162,4 @@
    public EntityAIControlledByPlayer getAIControlledByPlayer() {
       return this.aiControlledByPlayer;
    }
-
-   // $FF: synthetic method
-   public EntityAgeable createChild(EntityAgeable var1) {
-      return this.createChild(var1);
-   }
 }
