--- D:\CRAFTSTONE\DiffMakerPrivate\source\orig\net\minecraft\entity\player\EntityPlayer.java	2014-11-28 15:10:48.000000000 +0100
+++ D:\CRAFTSTONE\DiffMakerPrivate\source\new\net\minecraft\entity\player\EntityPlayer.java	2014-11-30 18:49:03.720203700 +0100
@@ -1,12 +1,15 @@
 package net.minecraft.entity.player;
 
+import com.craftstone.stone.entity.Player;
 import com.google.common.base.Charsets;
 import com.google.common.collect.Lists;
 import com.mojang.authlib.GameProfile;
+
 import java.util.Collection;
 import java.util.Iterator;
 import java.util.List;
 import java.util.UUID;
+
 import net.minecraft.block.Block;
 import net.minecraft.block.BlockBed;
 import net.minecraft.block.BlockDirectional;
@@ -79,7 +82,7 @@
 import net.minecraft.world.World;
 import net.minecraft.world.WorldSettings;
 
-public abstract class EntityPlayer extends EntityLivingBase {
+public abstract class EntityPlayer extends EntityLivingBase implements Player { //CraftStone
 
    public InventoryPlayer inventory = new InventoryPlayer(this);
    private InventoryEnderChest theInventoryEnderChest = new InventoryEnderChest();
@@ -1468,7 +1471,7 @@
 
    public abstract boolean func_175149_v();
 
-   public ItemStack[] getInventory() {
+   public ItemStack[] getVanillaInventory() {
       return this.inventory.armorInventory;
    }
 
@@ -1539,8 +1542,8 @@
       }
    }
 
-   public boolean sendCommandFeedback() {
-      return MinecraftServer.getServer().worldServers[0].getGameRules().getGameRuleBooleanValue("sendCommandFeedback");
+   public boolean sendCommandFeedback() { //TEMP
+      return MinecraftServer.getServer().worldServers.get(0).getGameRules().getGameRuleBooleanValue("sendCommandFeedback");
    }
 
    public boolean func_174820_d(int var1, ItemStack var2) {
