--- D:\CRAFTSTONE\DiffMakerPrivate\source\orig\net\minecraft\entity\EntityLiving.java	2014-11-28 15:10:44.000000000 +0100
+++ D:\CRAFTSTONE\DiffMakerPrivate\source\new\net\minecraft\entity\EntityLiving.java	2014-11-30 18:48:51.315425600 +0100
@@ -158,7 +158,7 @@
    protected int getExperiencePoints(EntityPlayer var1) {
       if(this.experienceValue > 0) {
          int var2 = this.experienceValue;
-         ItemStack[] var3 = this.getInventory();
+         ItemStack[] var3 = this.getVanillaInventory();
 
          for(int var4 = 0; var4 < var3.length; ++var4) {
             if(var3[var4] != null && this.equipmentDropChances[var4] <= 1.0F) {
@@ -531,12 +531,12 @@
       this.equipment[var1] = var2;
    }
 
-   public ItemStack[] getInventory() {
+   public ItemStack[] getVanillaInventory() {
       return this.equipment;
    }
 
    protected void dropEquipment(boolean var1, int var2) {
-      for(int var3 = 0; var3 < this.getInventory().length; ++var3) {
+      for(int var3 = 0; var3 < this.getVanillaInventory().length; ++var3) {
          ItemStack var4 = this.getEquipmentInSlot(var3);
          boolean var5 = this.equipmentDropChances[var3] > 1.0F;
          if(var4 != null && (var1 || var5) && this.rand.nextFloat() - (float)var2 * 0.01F < this.equipmentDropChances[var3]) {
