--- D:\CRAFTSTONE\DiffMakerPrivate\source\orig\net\minecraft\command\ICommand.java	2014-11-28 15:10:38.000000000 +0100
+++ D:\CRAFTSTONE\DiffMakerPrivate\source\new\net\minecraft\command\ICommand.java	2014-12-02 20:19:28.887571900 +0100
@@ -1,22 +1,29 @@
 package net.minecraft.command;
 
 import java.util.List;
+
 import net.minecraft.command.ICommandSender;
 import net.minecraft.util.BlockPos;
 
 public interface ICommand extends Comparable {
 
+   String getPermission(); //CraftStone
+	
+   String getPermissionMessage(); //CraftStone
+	
    String getCommandName();
 
    String getCommandUsage(ICommandSender var1);
 
-   List getCommandAliases();
+   String[] getCommandAliases(); //CraftStone String array
 
-   void processCommand(ICommandSender var1, String[] var2);
+   void processCommand(ICommandSender var1, String[] var2) throws WrongUsageException, NumberInvalidException, PlayerNotFoundException, CommandException;
 
    boolean canCommandSenderUseCommand(ICommandSender var1);
 
    List addTabCompletionOptions(ICommandSender var1, String[] var2, BlockPos var3);
 
    boolean isUsernameIndex(String[] var1, int var2);
+   
+   public String getDescription();
 }
