--- D:\CRAFTSTONE\DiffMakerPrivate\source\orig\net\minecraft\nbt\JsonToNBT.java	2014-11-28 15:10:54.000000000 +0100
+++ D:\CRAFTSTONE\DiffMakerPrivate\source\new\net\minecraft\nbt\JsonToNBT.java	2014-11-28 14:55:04.795311000 +0100
@@ -27,7 +27,7 @@
    private static final Pattern field_179273_b = Pattern.compile("\\[[-+\\d|,\\s]+\\]");
 
 
-   public static NBTTagCompound func_180713_a(String var0) {
+   public static NBTTagCompound func_180713_a(String var0) throws NBTException {
       var0 = var0.trim();
       if(!var0.startsWith("{")) {
          throw new NBTException("Invalid tag encountered, expected \'{\' as first char.");
@@ -38,7 +38,7 @@
       }
    }
 
-   static int func_150310_b(String var0) {
+   static int func_150310_b(String var0) throws NBTException {
       int var1 = 0;
       boolean var2 = false;
       Stack var3 = new Stack();
@@ -85,11 +85,11 @@
       }
    }
 
-   static JsonToNBT.Any func_179272_a(String ... var0) {
+   static JsonToNBT.Any func_179272_a(String ... var0) throws NBTException {
       return func_150316_a(var0[0], var0[1]);
    }
 
-   static JsonToNBT.Any func_150316_a(String var0, String var1) {
+   static JsonToNBT.Any func_150316_a(String var0, String var1) throws NBTException {
       var1 = var1.trim();
       String var3;
       boolean var4;
@@ -143,13 +143,13 @@
       }
    }
 
-   private static JsonToNBT.Any func_179270_a(String var0, boolean var1) {
+   private static JsonToNBT.Any func_179270_a(String var0, boolean var1) throws NBTException {
       String var2 = func_150313_b(var0, var1);
       String var3 = func_150311_c(var0, var1);
       return func_179272_a(new String[]{var2, var3});
    }
 
-   private static String func_150314_a(String var0, boolean var1) {
+   private static String func_150314_a(String var0, boolean var1) throws NBTException {
       int var2 = func_150312_a(var0, ':');
       int var3 = func_150312_a(var0, ',');
       if(var1) {
@@ -167,7 +167,7 @@
       return func_179269_a(var0, var2);
    }
 
-   private static String func_179269_a(String var0, int var1) {
+   private static String func_179269_a(String var0, int var1) throws NBTException {
       Stack var2 = new Stack();
       int var3 = var1 + 1;
       boolean var4 = false;
@@ -221,7 +221,7 @@
       return var0.substring(0, var3);
    }
 
-   private static String func_150313_b(String var0, boolean var1) {
+   private static String func_150313_b(String var0, boolean var1) throws NBTException {
       if(var1) {
          var0 = var0.trim();
          if(var0.startsWith("{") || var0.startsWith("[")) {
@@ -241,7 +241,7 @@
       }
    }
 
-   private static String func_150311_c(String var0, boolean var1) {
+   private static String func_150311_c(String var0, boolean var1) throws NBTException {
       if(var1) {
          var0 = var0.trim();
          if(var0.startsWith("{") || var0.startsWith("[")) {
