--- D:\CRAFTSTONE\DiffMakerPrivate\source\orig\net\minecraft\nbt\NBTTagString.java	2014-11-28 15:10:54.000000000 +0100
+++ D:\CRAFTSTONE\DiffMakerPrivate\source\new\net\minecraft\nbt\NBTTagString.java	2014-11-28 14:55:04.815311100 +0100
@@ -2,6 +2,8 @@
 
 import java.io.DataInput;
 import java.io.DataOutput;
+import java.io.IOException;
+
 import net.minecraft.nbt.NBTBase;
 import net.minecraft.nbt.NBTSizeTracker;
 
@@ -21,11 +23,11 @@
       }
    }
 
-   void write(DataOutput var1) {
+   void write(DataOutput var1) throws IOException {
       var1.writeUTF(this.data);
    }
 
-   void read(DataInput var1, int var2, NBTSizeTracker var3) {
+   void read(DataInput var1, int var2, NBTSizeTracker var3) throws IOException {
       this.data = var1.readUTF();
       var3.read((long)(16 * this.data.length()));
    }
