--- D:\CRAFTSTONE\DiffMakerPrivate\source\orig\net\minecraft\network\NetworkManager.java	2014-11-28 15:10:54.000000000 +0100
+++ D:\CRAFTSTONE\DiffMakerPrivate\source\new\net\minecraft\network\NetworkManager.java	2014-11-28 14:55:04.825311100 +0100
@@ -2,6 +2,7 @@
 
 import com.google.common.collect.Queues;
 import com.google.common.util.concurrent.ThreadFactoryBuilder;
+
 import io.netty.channel.Channel;
 import io.netty.channel.ChannelFuture;
 import io.netty.channel.ChannelFutureListener;
@@ -13,9 +14,12 @@
 import io.netty.channel.nio.NioEventLoopGroup;
 import io.netty.util.AttributeKey;
 import io.netty.util.concurrent.GenericFutureListener;
+
 import java.net.SocketAddress;
 import java.util.Queue;
+
 import javax.crypto.SecretKey;
+
 import net.minecraft.network.EnumConnectionState;
 import net.minecraft.network.EnumPacketDirection;
 import net.minecraft.network.INetHandler;
@@ -31,6 +35,7 @@
 import net.minecraft.util.CryptManager;
 import net.minecraft.util.IChatComponent;
 import net.minecraft.util.LazyLoadBase;
+
 import org.apache.commons.lang3.ArrayUtils;
 import org.apache.commons.lang3.Validate;
 import org.apache.logging.log4j.LogManager;
@@ -76,7 +81,7 @@
       this.direction = var1;
    }
 
-   public void channelActive(ChannelHandlerContext var1) {
+   public void channelActive(ChannelHandlerContext var1) throws Exception {
       super.channelActive(var1);
       this.channel = var1.channel();
       this.socketAddress = this.channel.remoteAddress();
