--- D:\CRAFTSTONE\DiffMakerPrivate\source\orig\net\minecraft\network\PacketBuffer.java	2014-11-28 15:10:56.000000000 +0100
+++ D:\CRAFTSTONE\DiffMakerPrivate\source\new\net\minecraft\network\PacketBuffer.java	2014-11-28 14:55:04.835311100 +0100
@@ -152,7 +152,7 @@
 
    }
 
-   public NBTTagCompound readNBTTagCompoundFromBuffer() {
+   public NBTTagCompound readNBTTagCompoundFromBuffer() throws IOException {
       int var1 = this.readerIndex();
       byte var2 = this.readByte();
       if(var2 == 0) {
@@ -180,7 +180,7 @@
 
    }
 
-   public ItemStack readItemStackFromBuffer() {
+   public ItemStack readItemStackFromBuffer() throws IOException {
       ItemStack var1 = null;
       short var2 = this.readShort();
       if(var2 >= 0) {
@@ -412,11 +412,11 @@
       return this.buf.getBytes(var1, var2);
    }
 
-   public ByteBuf getBytes(int var1, OutputStream var2, int var3) {
+   public ByteBuf getBytes(int var1, OutputStream var2, int var3) throws IOException {
       return this.buf.getBytes(var1, var2, var3);
    }
 
-   public int getBytes(int var1, GatheringByteChannel var2, int var3) {
+   public int getBytes(int var1, GatheringByteChannel var2, int var3) throws IOException {
       return this.buf.getBytes(var1, var2, var3);
    }
 
@@ -480,11 +480,11 @@
       return this.buf.setBytes(var1, var2);
    }
 
-   public int setBytes(int var1, InputStream var2, int var3) {
+   public int setBytes(int var1, InputStream var2, int var3) throws IOException {
       return this.buf.setBytes(var1, var2, var3);
    }
 
-   public int setBytes(int var1, ScatteringByteChannel var2, int var3) {
+   public int setBytes(int var1, ScatteringByteChannel var2, int var3) throws IOException {
       return this.buf.setBytes(var1, var2, var3);
    }
 
@@ -576,11 +576,11 @@
       return this.buf.readBytes(var1);
    }
 
-   public ByteBuf readBytes(OutputStream var1, int var2) {
+   public ByteBuf readBytes(OutputStream var1, int var2) throws IOException {
       return this.buf.readBytes(var1, var2);
    }
 
-   public int readBytes(GatheringByteChannel var1, int var2) {
+   public int readBytes(GatheringByteChannel var1, int var2) throws IOException {
       return this.buf.readBytes(var1, var2);
    }
 
@@ -648,11 +648,11 @@
       return this.buf.writeBytes(var1);
    }
 
-   public int writeBytes(InputStream var1, int var2) {
+   public int writeBytes(InputStream var1, int var2) throws IOException {
       return this.buf.writeBytes(var1, var2);
    }
 
-   public int writeBytes(ScatteringByteChannel var1, int var2) {
+   public int writeBytes(ScatteringByteChannel var1, int var2) throws IOException {
       return this.buf.writeBytes(var1, var2);
    }
 
@@ -799,22 +799,4 @@
    public boolean release(int var1) {
       return this.buf.release(var1);
    }
-
-   // $FF: synthetic method
-   // $FF: bridge method
-   public ReferenceCounted retain(int var1) {
-      return this.retain(var1);
-   }
-
-   // $FF: synthetic method
-   // $FF: bridge method
-   public ReferenceCounted retain() {
-      return this.retain();
-   }
-
-   // $FF: synthetic method
-   // $FF: bridge method
-   public int compareTo(Object var1) {
-      return this.compareTo((ByteBuf)var1);
-   }
 }
