--- D:\CRAFTSTONE\DiffMakerPrivate\source\orig\net\minecraft\network\play\server\S0FPacketSpawnMob.java	2014-11-28 15:10:56.000000000 +0100
+++ D:\CRAFTSTONE\DiffMakerPrivate\source\new\net\minecraft\network\play\server\S0FPacketSpawnMob.java	2014-11-28 14:55:04.845311100 +0100
@@ -1,6 +1,8 @@
 package net.minecraft.network.play.server;
 
+import java.io.IOException;
 import java.util.List;
+
 import net.minecraft.entity.DataWatcher;
 import net.minecraft.entity.EntityList;
 import net.minecraft.entity.EntityLivingBase;
@@ -72,7 +74,7 @@
       this.field_149043_l = var1.getDataWatcher();
    }
 
-   public void readPacketData(PacketBuffer var1) {
+   public void readPacketData(PacketBuffer var1) throws IOException {
       this.field_149042_a = var1.readVarIntFromBuffer();
       this.field_149040_b = var1.readByte() & 255;
       this.field_149041_c = var1.readInt();
