--- D:\CRAFTSTONE\DiffMakerPrivate\source\orig\net\minecraft\network\play\server\S49PacketUpdateEntityNBT.java	2014-11-28 15:10:58.000000000 +0100
+++ D:\CRAFTSTONE\DiffMakerPrivate\source\new\net\minecraft\network\play\server\S49PacketUpdateEntityNBT.java	2014-11-28 14:55:04.855311100 +0100
@@ -1,5 +1,7 @@
 package net.minecraft.network.play.server;
 
+import java.io.IOException;
+
 import net.minecraft.nbt.NBTTagCompound;
 import net.minecraft.network.INetHandler;
 import net.minecraft.network.Packet;
@@ -19,7 +21,7 @@
       this.field_179765_b = var2;
    }
 
-   public void readPacketData(PacketBuffer var1) {
+   public void readPacketData(PacketBuffer var1) throws IOException {
       this.field_179766_a = var1.readVarIntFromBuffer();
       this.field_179765_b = var1.readNBTTagCompoundFromBuffer();
    }
